### PR TITLE
Implement #546 (Block object)

### LINKED ISSUE #546: Support block object
Currently, it's not possible to pass around (and execute) block object. This is a very useful feature

### PR DESCRIPTION
**Pull request by st0012**
This closes #546

### CODE REVIEW

File: vm/evaluation_test.go [hunk 1]
```diff
@@ -1509,6 +1509,44 @@ func TestMethodCallWithBlockArgument(t *testing.T) {
 			  ten + 20
 			end
 			`, 30},
+		// Get Block
+		{`
```
Comment (st0012) [inline vm/evaluation_test.go]: @saveriomiroddi Are these what you need to implement #537?

File: vm/block_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import "testing"
+
+func TestBlockInitialize(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		{`
+b = Block.new do
```
Comment (st0012) [inline vm/block_test.go]: I also implemented something like `Proc.new`. But I always don't think `Proc` is a good name, so I just call it `Block`.
And I **won't** support something like `proc`.

File: vm/block_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  100
+end
+
+b.call
+`, 100},
+
+		{`
```
Comment (st0012) [inline vm/block_test.go]: @saveriomiroddi @hachi8833 Do you think we need more complex cases like this?

Comment (hachi8833) [reply vm/block_test.go]: Looks sufficient by now.

**Comment by st0012** [comment]
@saveriomiroddi We already have `block_given?` for a while, but it exists as a method instead a keyword.
I'll add more test cases for both `get_block` and `block_given?`.

File: vm/evaluation_test.go (hunk 1 above)
Comment (saveriomiroddi) [reply vm/evaluation_test.go]: > Are these what you need to implement #537?

@st0012 yes, I think so!

**Pull request merged**